lib/engine/part/partition.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+require_relative 'base'
+
+module Engine
+  module Part
+    class Partition < Base
+      attr_reader :a, :a_sign, :b, :b_sign, :type, :restrict, :magnet, :blockers, :inner, :outer
+
+      SIGN = {
+        '-' => -1,
+        nil => 0,
+        '+' => 1,
+      }.freeze
```

Comment:
why do you need sign?

Reply:
It's only a rendering thing. It draws the line a little bit to one edge or the other. It's important because the hex needs some information about what the river is going to avoid doing sharp corners (and I thought the sign was a simpler solution then having the code actually look at the bordering hexes)